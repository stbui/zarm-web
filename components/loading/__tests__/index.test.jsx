import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../index';

describe('Loading', () => {
  it('renders Loading of different size correctly', () => {
    const wrapper = render(
      <>
        <Loading size="lg" visible />
        <Loading size="md" visible />
        <Loading size="sm" visible />
        <Loading size="xs" visible />
        <Loading size="lg" visible>lg尺寸</Loading>
        <Loading size="md" visible>md尺寸</Loading>
        <Loading size="sm" visible>sm尺寸</Loading>
        <Loading size="xs" visible>xs尺寸</Loading>
      </>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders hidden Loading correctly', async () => {
    const wrapper = mount(
      <Loading visible={false} />,
    );

    expect(wrapper.exists('.zw-loading')).toEqual(false);
    wrapper.setProps({ children: 'hidden loading' });
    expect(wrapper.exists('.zw-loading')).toEqual(true);
  });

  it('renders text Loading correctly', () => {
    const wrapper = mount(
      <Loading visible text="这里是text文案..." />,
    );

    expect(wrapper.find('.zw-loading__text').text()).toEqual('这里是text文案...');
  });

  it('renders Loading useing custom indicator', () => {
    const customIndicator = <div className="custom-indicator" />;
    const wrapper = mount(
      <Loading visible indicator={customIndicator} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders delay Loading correctly', async () => {
    const wrapper = mount(
      <Loading delay={300} visible text="这里是text文案..." />,
    );

    expect(wrapper.exists('.zw-loading')).toEqual(false);
    await new Promise((resolve) => setTimeout(resolve, 500));
    wrapper.update();
    expect(wrapper.exists('.zw-loading')).toEqual(true);
    wrapper.unmount();
  });

  it('renders fullscreen Loading correctly', () => {
    const wrapper = mount(
      <Loading fullscreen visible>全屏显示</Loading>,
    );

    expect(wrapper.exists('.zw-loading--fullscreen')).toBe(true);
    wrapper.setProps({ text: '全屏text提示' });
    expect(wrapper.find('.zw-loading__text').text()).toEqual('全屏text提示');
  });
});
