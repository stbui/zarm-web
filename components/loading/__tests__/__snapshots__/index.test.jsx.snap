// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders Loading of different size correctly 1`] = `
Array [
  <div
    class="zw-loading zw-loading--lg zw-loading--active zw-loading--inline"
  >
    <span
      class="zw-activity-indicator zw-activity-indicator--circular"
    >
      <svg
        viewBox="31 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </span>
  </div>,
  <div
    class="zw-loading zw-loading--md zw-loading--active zw-loading--inline"
  >
    <span
      class="zw-activity-indicator zw-activity-indicator--circular"
    >
      <svg
        viewBox="31 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </span>
  </div>,
  <div
    class="zw-loading zw-loading--sm zw-loading--active zw-loading--inline"
  >
    <span
      class="zw-activity-indicator zw-activity-indicator--circular"
    >
      <svg
        viewBox="31 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </span>
  </div>,
  <div
    class="zw-loading zw-loading--xs zw-loading--active zw-loading--inline"
  >
    <span
      class="zw-activity-indicator zw-activity-indicator--circular"
    >
      <svg
        viewBox="31 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </span>
  </div>,
  <div
    class="zw-loading zw-loading--lg zw-loading--active"
  >
    <div
      class="zw-loading__container"
    >
      lg尺寸
    </div>
    <div
      class="zw-loading__spin__wrapper"
    >
      <div
        class="zw-loading__spin"
      >
        <span
          class="zw-activity-indicator zw-activity-indicator--circular"
        >
          <svg
            viewBox="31 31 62 62"
          >
            <circle
              cx="62"
              cy="62"
              fill="none"
              r="28.5"
              style="stroke-width:5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  <div
    class="zw-loading zw-loading--md zw-loading--active"
  >
    <div
      class="zw-loading__container"
    >
      md尺寸
    </div>
    <div
      class="zw-loading__spin__wrapper"
    >
      <div
        class="zw-loading__spin"
      >
        <span
          class="zw-activity-indicator zw-activity-indicator--circular"
        >
          <svg
            viewBox="31 31 62 62"
          >
            <circle
              cx="62"
              cy="62"
              fill="none"
              r="28.5"
              style="stroke-width:5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  <div
    class="zw-loading zw-loading--sm zw-loading--active"
  >
    <div
      class="zw-loading__container"
    >
      sm尺寸
    </div>
    <div
      class="zw-loading__spin__wrapper"
    >
      <div
        class="zw-loading__spin"
      >
        <span
          class="zw-activity-indicator zw-activity-indicator--circular"
        >
          <svg
            viewBox="31 31 62 62"
          >
            <circle
              cx="62"
              cy="62"
              fill="none"
              r="28.5"
              style="stroke-width:5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
  <div
    class="zw-loading zw-loading--xs zw-loading--active"
  >
    <div
      class="zw-loading__container"
    >
      xs尺寸
    </div>
    <div
      class="zw-loading__spin__wrapper"
    >
      <div
        class="zw-loading__spin"
      >
        <span
          class="zw-activity-indicator zw-activity-indicator--circular"
        >
          <svg
            viewBox="31 31 62 62"
          >
            <circle
              cx="62"
              cy="62"
              fill="none"
              r="28.5"
              style="stroke-width:5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`Loading renders Loading useing custom indicator 1`] = `
<Loading
  indicator={
    <div
      className="custom-indicator"
    />
  }
  prefixCls="zw-loading"
  size="md"
  visible={true}
>
  <div
    className="zw-loading zw-loading--md zw-loading--active zw-loading--inline"
  >
    <div
      className="custom-indicator"
    />
  </div>
</Loading>
`;
